File: starter_kits/Go/src/hlt/Game.go
package hlt

import (
	"fmt"

	"./gameconfig"
	"./input"
	"./log"
)

// Game - Structure holding all metadata for the game
type Game struct {
	NumPlayers int
	Me         *Player
	Players    []*Player
	Map        *GameMap
	TurnNumber int
}

func (g *Game) String() string {
	return fmt.Sprintf("Game{NumPlayers=%d,Me=%s,Players=%d,Map=%s}", g.NumPlayers, g.Me.String(), len(g.Players), g.Map.String())
}

// Ready - When run, notifies the server that the bot is ready to start
func (g *Game) Ready(name string) {
	fmt.Println(name)
}

// NewGame - Creates a new game for bot named by the parameter
func NewGame() *Game {
	var input = input.GetInstance()
	if !input.Scanner.Scan() {
		return nil
	}

	var constantsString = input.Scanner.Text()
	gameconfig.Init(constantsString)
	var NumPlayers, _ = input.GetInt()
	var myID, _ = input.GetInt()
	var Players = make([]*Player, NumPlayers)
	for i := range Players {
		Players[i] = NewPlayer()
	}
	var gameMap = GenerateGameMap()
	var me = Players[myID]
	return &Game{NumPlayers, me, Players, gameMap, 0}
}

// UpdateFrame - Runs a single turn in the game
func (g *Game) UpdateFrame() {
	var logger = log.GetInstance()
	var input = input.GetInstance()
	g.TurnNumber, _ = input.GetInt()
	logger.Printf("=============== TURN %d ================\n", g.TurnNumber)
	for i := range g.Players {
		// Player ID variable being read here. It isn't used, so we can't assign it in Go without error, but we still need to consume it from the input
		input.GetInt()
		var numShips, _ = input.GetInt()
		var numDropoffs, _ = input.GetInt()
		var halite, _ = input.GetInt()
		g.Players[i].Update(numShips, numDropoffs, halite)
	}
	g.Map.Update()
	for i := range g.Players {
		var player = g.Players[i]
		for j := range player.Ships {
			var ship = player.Ships[j]
			g.Map.AtEntity(ship.E).MarkUnsafe(ship)
		}
		for k := range player.Dropoffs {
			var dropoff = player.Dropoffs[k]
			g.Map.AtEntity(dropoff.E).structure = dropoff.E
		}
	}
}

// EndTurn -
func (g *Game) EndTurn(commands []Command) {
	for _, command := range commands {
		fmt.Print(command.CommandString())
		fmt.Print(" ")
	}
	fmt.Println()
}

File: starter_kits/C#/Core/Game.cs
using System;
using System.Collections.Generic;

namespace Halite3.Core
{
    public class Game
    {
        public int TurnNumber { get; private set; }
        public PlayerId MyId { get; private set; }
        public Dictionary<int, Player> Players { get; private set; }
        public Player Me { get; private set; }
        public GameMap GameMap { get; private set; }

        public Game()
        {
            Players = new Dictionary<int, Player>();

            Constants.populateConstants(Input.ReadLine());

            Input input = Input.readInput();
            int numPlayers = input.getInt();
            MyId = new PlayerId(input.getInt());

            Log.logger().Info("My Id: " + MyId.Id);

            for (var i = 0; i < numPlayers; ++i)
            {
                var player = Player._generate();
                Players.Add(player.Id.Id, player);
            }

            Me = Players[MyId.Id];
            GameMap = GameMap._generate();
        }

        public void Ready(string name)
        {
            Console.WriteLine(name);
        }

        public void UpdateFrame()
        {
            TurnNumber = Input.readInput().getInt();
            Log.logger().Info("=============== TURN " + TurnNumber + " ================");

            for (int i = 0; i < Players.Count; ++i)
            {
                Input input = Input.readInput();

                PlayerId currentPlayerId = new PlayerId(input.getInt());
                int numShips = input.getInt();
                int numDropoffs = input.getInt();
                int halite = input.getInt();

                Players[currentPlayerId.Id]._update(numShips, numDropoffs, halite);
            }

            GameMap._update();

            foreach (var player in Players.Values)
            {
                foreach (var ship in player.Ships.Values)
                {
                    GameMap.At(ship).markUnsafe(ship);
                }

                GameMap.At(player.Shipyard).Structure = player.Shipyard;

                foreach (var dropoff in player.Dropoffs.Values)
                {
                    GameMap.At(dropoff).Structure = dropoff;
                }
            }
        }

        public void EndTurn(List<Command> commands)
        {
            foreach (var command in commands)
            {
                Console.WriteLine(command.CommandValue);
                Console.WriteLine(' ');
            }
            Console.WriteLine();
        }
    }
}